### PR TITLE
Added os name and os version

### PR DESCRIPTION
**Pull request by miniche**
Getting os name and os version in Collectors and display it on web pages, like this : 

Capture d e cran 2013-01-06 a 22 56 54

### CODE REVIEW

File: centralreport/cr/collectors.py [hunk 1]
```diff
@@ -268,9 +272,38 @@ def get_infos(self):
...
+
+        if Config.HOST_CURRENT == Config.HOST_DEBIAN:
+            os_name = 'Debian'
+            os_version = subprocess.Popen(['cat', '/etc/debian_version'], stdout=subprocess.PIPE, close_fds=True).communicate()[0]
+        elif Config.HOST_CURRENT == Config.HOST_UBUNTU:
+            os_name = 'Ubuntu'
+
+            # OS version for Ubuntu
+            os_version = 'Unkown'
+            os_version_full = subprocess.Popen(['cat', '/etc/lsb-release'], stdout=subprocess.PIPE, close_fds=True).communicate()[0]
```
Comment (Guimove) [inline centralreport/cr/collectors.py:288]: You can use this file for all Linux Distrib and parse it to get the version

Comment (miniche) [inline centralreport/cr/collectors.py:288]: On a basic Debian (6.0.5), /etc/lsb-release file not exist ;-)

Comment (Guimove) [inline centralreport/cr/collectors.py:288]: Sure ? It exist on mine 6.0.6 and 6.0.3 at my internship

Comment (miniche) [inline centralreport/cr/collectors.py:288]: On my vm with 6.0.5 and on a physical server with 6.0.1, i don't have the lsb-release file, only the "lsb_release" command.

Maybe specifics packages are installed on your servers, no ?

File: centralreport/cr/collectors.py [hunk 2]
```diff
@@ -268,9 +272,38 @@ def get_infos(self):
...
+        os_version = ''
+
+        if Config.HOST_CURRENT == Config.HOST_DEBIAN:
+            os_name = 'Debian'
+            os_version = subprocess.Popen(['cat', '/etc/debian_version'], stdout=subprocess.PIPE, close_fds=True).communicate()[0]
+        elif Config.HOST_CURRENT == Config.HOST_UBUNTU:
+            os_name = 'Ubuntu'
+
+            # OS version for Ubuntu
+            os_version = 'Unkown'
```
Comment (Ninir) [inline centralreport/cr/collectors.py]: Typo: Unknown

File: centralreport/cr/collectors.py [hunk 3]
```diff
@@ -268,9 +272,38 @@ def get_infos(self):
...
+            os_version_full = subprocess.Popen(['cat', '/etc/lsb-release'], stdout=subprocess.PIPE, close_fds=True).communicate()[0]
+            os_version_lines = os_version_full.splitlines()
+
+            # Looking for "DISTRIB_RELEASE" key
+            for i in range(0,len(os_version_lines)):
```
Comment (Ninir) [inline centralreport/cr/collectors.py]: Missing space before `len(...)`

File: centralreport/cr/collectors.py [hunk 4]
```diff
@@ -268,9 +272,38 @@ def get_infos(self):
...
+
+                if os_version_lines[i].startswith("DISTRIB_RELEASE"):
+                    try:
+                        os_version = os_version_lines[i].split("=")[1].replace(' ','')
```
Comment (Ninir) [inline centralreport/cr/collectors.py]: Missing space before `,''`

File: centralreport/web/tpl/index.tpl [hunk 5]
```diff
@@ -3,10 +3,34 @@
...
+
+    <div class="headhand clearfix">
+        Last check : <span id="last_check_date">{{ last_check }}</span>
+        <small id="ajax_enabled"></small>
+    </div>
+
+    <div id="host_header">
+        <div class="row-fluid">
+            <div class="span1">
+                {% if host_os == 'MAC' %}
```
Comment (Ninir) [inline centralreport/web/tpl/index.tpl]: Please swap arguments ;) 

http://www.codinghorror.com/blog/2012/07/new-programming-jargon.html

File: centralreport/cr/collectors.py [hunk 6]
```diff
@@ -268,9 +272,38 @@ def get_infos(self):
...
+                    except:
+                        crLog.writeError('Error getting Ubuntu version')
+                        os_version = ''
+
+
+
```
Comment (Ninir) [inline centralreport/cr/collectors.py]: 2 additional lines here

File: centralreport/cr/collectors.py (hunk 3 above)
Comment (miniche) [inline centralreport/cr/collectors.py]: Ninir is back from holidays :D ;-)

File: centralreport/cr/collectors.py (hunk 1 above)
Comment (Guimove) [inline centralreport/cr/collectors.py:288]: Maybe : Apache, Php, Mysql, Htop, zsh...
So if you don't have it, it's ok for me

**Comment by Ninir** [comment]
You have 2 PNG without a transparent background. You should either set every background to white and then use JPG files, or add transparency to be consistent with the other one.

**Comment by miniche** [comment]
You're right, @Ninir. I'll watch it tomorrow ;-)

File: centralreport/cr/collectors.py (hunk 3 above)
Comment (Ninir) [inline centralreport/cr/collectors.py]: At your service :p

**Comment by miniche** [comment]
Good for me. And you ?

File: centralreport/web/server.py [hunk 7]
```diff
@@ -111,7 +111,18 @@ def index(self):
...
+        # Host informations
         tmpl_vars['hostname'] = Checks.hostEntity.hostname
+        tmpl_vars['os_name'] = Checks.hostEntity.osName
+        tmpl_vars['os_version'] = Checks.hostEntity.osVersion
+
+        if Config.HOST_CURRENT == Config.HOST_MAC:
+            tmpl_vars['host_os'] = 'MAC'
+        elif Config.HOST_CURRENT == Config.HOST_UBUNTU:
+            tmpl_vars['host_os'] = 'UBUNTU'
+        elif Config.HOST_DEBIAN == Config.HOST_DEBIAN:
```
Comment (Ninir) [inline centralreport/web/server.py]: `Config.HOST_DEBIAN == Config.HOST_DEBIAN`?

It seems you are not testing the good (left) part :P

File: centralreport/web/tpl/index.tpl [hunk 8]
```diff
@@ -3,10 +3,34 @@
...
+    <div class="headhand clearfix">
+        Last check : <span id="last_check_date">{{ last_check }}</span>
+        <small id="ajax_enabled"></small>
+    </div>
+
+    <div id="host_header">
+        <div class="row-fluid">
+            <div class="span1">
+                {% if 'MAC' == host_os %}
+                    <img src="img/logos/Apple_logo.png" alt="Apple" />
```
Comment (Ninir) [inline centralreport/web/tpl/index.tpl]: I suggest to suffix the alt attribute with "logo" in order to have "Apple logo", "Ubuntu logo", etc.

File: centralreport/cr/collectors.py [hunk 9]
```diff
@@ -268,9 +272,36 @@ def get_infos(self):
...
+            os_version = subprocess.Popen(['cat', '/etc/debian_version'], stdout=subprocess.PIPE, close_fds=True).communicate()[0]
+        elif Config.HOST_UBUNTU == Config.HOST_CURRENT:
+            os_name = 'Ubuntu'
+
+            # OS version for Ubuntu
+            os_version = 'Unknown'
+            os_version_full = subprocess.Popen(['cat', '/etc/lsb-release'], stdout=subprocess.PIPE, close_fds=True).communicate()[0]
+            os_version_lines = os_version_full.splitlines()
+
+            # Looking for "DISTRIB_RELEASE" key
```
Comment (Ninir) [inline centralreport/cr/collectors.py]: Looking for `THE` "DISTRIB_RELEASE" key

**Pull request merged**